src/delta.rs

```diff
@@ -147,7 +148,11 @@ impl<'a> StateMachine<'a> {
     where
         I: BufRead,
     {
+        let mut debug_helper = RecordDeltaCall::new(self.config);
+
         while let Some(Ok(raw_line_bytes)) = lines.next() {
+            debug_helper.write(raw_line_bytes);
+
```

Comment:
So IIUC, the pattern we're using is

Create the `debug_helper = RecordDeltaCall::new()`; This causes some default diagnostics to be written.

...
Append additional arbitrary diagnostics via `debug_helper.write`
...
Terminate the file in the `Drop` impl for `RecordDeltaCall`.


But if the `DELTA_DEBUG_LOGFILE` is not set, then, although we do a few function calls, they are all no-ops.

Reply:
Indeed, `write()` will usually return after an an easily branch-predicted if.